src/components/PaginatedTable/requestBatcher.ts

```diff
@@ -0,0 +1,182 @@
+                currentGroup = [sorted[i]];
+            }
+        }
+
+        // Add the last group
+        groups.push(this.createBatchGroup(currentGroup));
+
+        return groups;
+    }
+
+    private createBatchGroup<T, F>(requests: QueuedRequest<T, F>[]): BatchGroup<T, F> {
+        const minOffset = Math.min(...requests.map((r) => r.params.offset));
+        const maxOffset = Math.max(...requests.map((r) => r.params.offset));
+        const limit = requests[0].params.limit;
+        const totalLimit = requests.length * limit;
+
+        return {
+            requests,
+            batchKey: this.createBatchKey(requests[0].params),
+            minOffset,
+            maxOffset,
+            totalLimit,
+        };
+    }
+
+    private async executeBatch<T, F>(group: BatchGroup<T, F>): Promise<void> {
+        const firstRequest = group.requests[0];
+        const batchParams = {
+            ...firstRequest.params,
+            offset: group.minOffset,
+            limit: group.totalLimit,
+        };
+
+        try {
+            const response = await firstRequest.params.fetchData({
+                limit: batchParams.limit,
+                offset: batchParams.offset,
+                filters: batchParams.filters,
+                sortParams: batchParams.sortParams,
+                columnsIds: batchParams.columnsIds,
+                signal: firstRequest.signal,
+            });
+
+            // Split the response data among individual requests
+            this.splitAndDistributeResponse(group, response);
+        } catch (error) {
+            // If batch fails, reject all requests in the group
+            group.requests.forEach((request) => {
+                request.resolve({error});
+            });
+        }
+    }
+
+    private splitAndDistributeResponse<T, F>(
+        group: BatchGroup<T, F>,
+        batchResponse: PaginatedTableData<T>,
+    ): void {
+        const limit = group.requests[0].params.limit;
+
+        group.requests.forEach((request, index) => {
+            const startIndex = index * limit;
+            const endIndex = startIndex + limit;
+            const chunkData = batchResponse.data.slice(startIndex, endIndex);
+
+            const chunkResponse: PaginatedTableData<T> = {
+                ...batchResponse,
+                data: chunkData,
+                total: batchResponse.total,
+                found: batchResponse.found,
+            };
+
+            request.resolve({data: chunkResponse});
+        });
+    }
+
+    private async processBatch(): Promise<void> {
+        const allQueues = Array.from(this.requestQueue.entries());
+        this.requestQueue.clear();
+        this.batchTimeout = null;
+
+        for (const [_batchKey, requests] of allQueues) {
+            const groups = this.groupConsecutiveRequests(requests);
+
+            // Execute each group (consecutive chunks) as a separate batch
+            await Promise.all(groups.map((group) => this.executeBatch(group)));
```

Comment:
I suppose it should be allSettled here - if one group fails it doesnt mean everybody should fail

Reply:
there is try catch block inside executeBatch - so I suppose this case is already handled